pandas/core/dtypes/common.py

```diff
@@ -2049,10 +2049,8 @@ def pandas_dtype(dtype):
     # raise a consistent TypeError if failed
     try:
         npdtype = np.dtype(dtype)
-    except Exception:
-        # we don't want to force a repr of the non-string
-        if not isinstance(dtype, str):
-            raise TypeError("data type not understood")
+    except SyntaxError:
+        # np.dtype uses `eval` which can raise SyntaxError
```

Comment:
You can easily end up here with something that does not raise a syntaxError

But I assume the idea is that in that case the TypeError generated by numpy is fine to raise directly? 

If so, can you add a comment for that?

Reply:
(based on your comment below, I suppose that is indeed the case)

Reply:
@jorisvandenbossche I am not sure it matters; this will raise a TypeError if its invalid anyhow (catching the SyntaxError is just for hygiene of the error message)